service/service.go

```diff
@@ -35,16 +35,14 @@ const (
 // to depend on the client. The code pertaining specifically to the client side and flow side
 // configs should be kept in the client/flow repos, and addresses and tls values should be
 // passed as arguments.
-func GrpcConnection(client, server string) (*grpc.ClientConn, error) {
-
+func GrpcConnection(client, server string, insecure bool) (*grpc.ClientConn, error) {
```

Comment:
Pretty sure this means it will only work on insecure connections if set to true right? ie. Production and staging won't work?

I think the behaviour we want is to always try secure connections, but only fallback to insecure connections if this is true.